### PR TITLE
CUMULUS-2692 -- Initial procedure commit and migration update

### PR DESCRIPTION
**Pull request by Jkovarik**
**Summary:** Summary of changes

Addresses [CUMULUS-2962](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2962)

## Changes

* Re-added files index
* Added procedure for migration

## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+  ```text
+  select pid, query, state, wait_event_type, wait_event from pg_stat_activity where state = 'active';
+  ```
+
+  If query rows are returned with a `query` value that involves the files table,
+  make sure ingest is halted and no other granule-update activity is running on
+  the system.
+
+  Note: In rare instances if there are hung queries that are unable to resolve, it may be necessary to
+  manually use psql `Server Signaling Functions` `pg_cancel_backend` and/or
```
Comment (Jkovarik) [inline CHANGELOG.md:64]: Include doc link

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### MIGRATION NOTES
 
+- The changes introduced in CUMULUS-2962 will re-introduce a
+  `files_granules_cumulus_id_index` on the `files` table in the RDS database.
+  This index will be automatically created as part of the bootstrap lambda
+  function *on deployment* of the `data-persistence` module.
+
+  *In cases where the index is already applied, this update have no effect*.
```
Comment (npauzenga) [inline CHANGELOG.md:16]:
```suggestion
  *In cases where the index is already applied, this update will have no effect*.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+
+  **Please Note**: In some cases where ingest is occurring at high volume levels and/or there are
+  record volumes in the files table of > 150M files records, the migration may
+  fail on deployment due to timing required to both acquire the table state needed for the
+  migration and time to create the index given the resources available.
```
Comment (npauzenga) [inline CHANGELOG.md:21]:
```suggestion
  **Please Note**: In some cases where ingest is occurring at high volume levels and/or the
  files table has > 150M file records, the migration may
  fail on deployment due to timing required to both acquire the table state needed for the
  migration and time to create the index given the resources available.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+
+  For reference a rx.5 large Aurora/RDS database
+  with no activity took roughly 6 minutes to create the index for a 300M file
+  table with no ingest, however timed out when the same migration was attempted
+  in production with possible activity on the table.
```
Comment (npauzenga) [inline CHANGELOG.md:26]:
```suggestion
  with *no activity* took roughly 6 minutes to create the index for a file table with 300M records and no active ingest, however timed out when the same migration was attempted
  in production with possible activity on the table.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,107 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+  `pg_terminate_backend` if the migration will not complete in the next step.
+
+  - Create the Index
+
+  Run the following query to create the index.    Depending on the situation
+  this may take many minutes to complete, and you will note your CPU load and
+  disk I/O rates increase on your cluster:
+
+  ```text
+  CREATE INDEX files_granule_cumulus_id_index ON files (granule_cumulus_id);
```
Comment (npauzenga) [inline CHANGELOG.md:74]: Is there any value in `IF NOT EXISTS` if they're running it manually? Maybe if they're running this by mistake and have already added the index?

**Review by npauzenga — CHANGES_REQUESTED** [review]
Looks great @Jkovarik! I added a couple suggestions just for kicks. Take 'em or leave 'em. The only actual question was the one about `IF NOT EXISTS` on the manual index creation instructions.

File: CHANGELOG.md (hunk 5 above)
Comment (npauzenga) [reply CHANGELOG.md:74]: I may have just missed this point in our convo 🤔

Comment (Jkovarik) [reply CHANGELOG.md:73]: If it *does* exist, it does something like: 

---
jk_tf4_db=> CREATE INDEX files_granule_cumulus_id_index ON files (granule_cumulus_id)                                                                                                                                     ;
ERROR:  relation "files_granule_cumulus_id_index" already exists
---

Which in the case of manual creation is probably.... useful?

**Review by npauzenga — APPROVED** [review]
👍

**Pull request merged**